File: API-MS_28jun2023/LUCAS BORGES DA SILVA/api semestral - lucas tribal/areas.py

def main():
    
    opcoes_validas = ['Q','q','R','r','T','t',]
    print (f"As figuras disponíveis são : Quadrado,Retângulo,Triângulo,Trapézio e círculo")

    primeira_figura = obter_texto("Digite a letra inicial de sua figura (Digite P para trapézio): ")

    while primeira_figura != 'Q' and primeira_figura != 'q' and primeira_figura !='R' and primeira_figura != 'r' and primeira_figura != 'T' and primeira_figura != 'P' and primeira_figura !='p' and primeira_figura != 'C' and primeira_figura != "C":
        print("Figura inválida! Digite uma letra válida!")
        primeira_figura = obter_texto("Digite a letra inicial de sua figura (Digite P para trapézio): ")

    segunda_figura = obter_texto("Digite a letra inicial de sua figura (Digite P para trapézio): ")

    while segunda_figura != 'Q' and segunda_figura != 'q' and segunda_figura !='R' and segunda_figura != 'r' and segunda_figura != 'T' and segunda_figura != 'P' and segunda_figura !='p' and segunda_figura != 'C' and segunda_figura != "C":
        print("Figura inválida! Digite uma letra válida!")
        primeira_figura = obter_texto("Digite a letra inicial de sua figura (Digite P para trapézio): ")


    while primeira_figura == segunda_figura: 
        print("As duas figuras não podem ser do mesmo tipo! Digite novamente !")
        primeira_figura = obter_texto("Digite a letra inicial de sua figura (Digite P para trapézio): ")
        segunda_figura = obter_texto("Digite a letra inicial de sua figura: Digite P para trapézio): ")


    area_figura_1 = calcular_area_1(primeira_figura)
    area_figura_2 = calcular_area_2(segunda_figura)

    tipo_figura_1 = verificar_tipo_figura_1(primeira_figura)
    tipo_figura_2 = verificar_tipo_figura_1(primeira_figura)
    percentual_entre_areas = proporcao_percentual_das_areas(area_figura_1,area_figura_2)

    print(f"A figura 1 é um {tipo_figura_1} e sua área é de {area_figura_1:.2f}")
    print(f"A figura 2 é um {tipo_figura_2} e sua área é de {area_figura_2:.2f}")

    if area_figura_1>area_figura_2:
        print(f"A maior área é a da figura 1")
    else:
        print(f"A maior área é a da figura 2")

    
    print(f"A proporção  percentual entre a área da menor figura sobre a área da maior é de {percentual_entre_areas:.2f} ")
     








        
def proporcao_percentual_das_areas(area_figura_1,area_figura_2):
    if area_figura_1>area_figura_2:
        percentual = (area_figura_2/area_figura_1)*100
    else:
        percentual = (area_figura_1/area_figura_2)*100
    return percentual
        


            
def calcular_area_1(primeira_figura):
        
        if primeira_figura == 'Q' or primeira_figura == 'q':
            figura = "Quadrado"
            lado_quadrado = obter_numero_positivo("Digite o valor do lado de seu quadrado: ")
            area1 = calcular_area_quadrado(lado_quadrado)

        elif primeira_figura == 'R' or primeira_figura == 'r':
            figura = "Retângulo"
            comprimento_retangulo =obter_numero_positivo("Digite o valor do comprimento de seu retângulo: ")
            largura_retangulo = obter_numero_positivo("Digite o valor da largura de seu retângulo: ")
            area1 = calcular_area_retangulo(comprimento_retangulo,largura_retangulo)

        elif primeira_figura == 'T' or primeira_figura == 't':
            figura = "Triângulo"
            base_triangulo = obter_numero_positivo("Digite a base do triângulo: ")
            altura = obter_numero_positivo("Digite a altura do triângulo")
            area1 = calcular_area_triangulo(base_triangulo,altura)

        elif primeira_figura == 'P' or primeira_figura == 'p':
            base_maior_trapézio = obter_numero_positivo("Digite a base maior do trapézio: ")
            base_menor_trapézio = obter_numero_positivo("Digite a base menor do trapézio: ")
            altura_trapezio = obter_numero_positivo("Digite a altura do triângulo: ") 
            area1 = calcular_area_trapezio(base_maior_trapézio,base_menor_trapézio,altura_trapezio)
            
        elif primeira_figura == 'C' or primeira_figura == 'c':
            pi = 3.14159
            raio = obter_numero_positivo("Digite o raio do círculo: ")
            area1 = calcular_area_circulo(pi,raio)
        return area1


def calcular_area_2(segunda_figura):
        
        if segunda_figura == 'Q' or segunda_figura == 'q':
            figura = "Quadrado"
            lado_quadrado = obter_numero_positivo("Digite o valor do lado de seu quadrado: ")
            area2 = calcular_area_quadrado(lado_quadrado)

        elif segunda_figura == 'R' or segunda_figura == 'r':
            figura = "Retângulo"
            comprimento_retangulo =obter_numero_positivo("Digite o valor do comprimento de seu retângulo: ")
            largura_retangulo = obter_numero_positivo("Digite o valor da largura de seu retângulo: ")
            area2 = calcular_area_retangulo(comprimento_retangulo,largura_retangulo)

        elif segunda_figura == 'T' or segunda_figura == 't':
            figura = "Triângulo"
            base_triangulo = obter_numero_positivo("Digite a base do triângulo: ")
            altura = obter_numero_positivo("Digite a altura do triângulo")
            area2 = calcular_area_triangulo(base_triangulo,altura)

        elif segunda_figura == 'P' or segunda_figura == 'p':
            base_maior_trapézio = obter_numero_positivo("Digite a base maior do trapézio: ")
            base_menor_trapézio = obter_numero_positivo("Digite a base menor do trapézio: ")
            altura_trapezio = obter_numero_positivo("Digite a altura do triângulo: ") 
            area2 = calcular_area_trapezio(base_maior_trapézio,base_menor_trapézio,altura_trapezio)
            
        elif segunda_figura == 'C' or segunda_figura == 'c':
            pi = 3.14159
            raio = obter_numero_positivo("Digite o raio do círculo: ")
            area2 = calcular_area_circulo(pi,raio)
        return area2

def verificar_tipo_figura_1(primeira_figura):
        if primeira_figura == 'Q' or primeira_figura == 'q':
            figura = "Quadrado"
            return figura
        elif primeira_figura == 'R' or primeira_figura == 'r':
            figura = "Retângulo"
            return figura
        elif primeira_figura == 'T' or primeira_figura == 't':
            figura = "Triângulo"
            return figura
        elif primeira_figura == 'P' or primeira_figura == 'p':
            figura = "Trapézio"
            return figura
        elif primeira_figura == 'C' or primeira_figura == 'c':
            figura = "Círculo"
            return figura
        

def verificar_tipo_figura_2(segunda_figura):
        if segunda_figura == 'Q' or segunda_figura == 'q':
            figura = "Quadrado"
            return figura
        elif segunda_figura == 'R' or segunda_figura == 'r':
            figura = "Retângulo"
            return figura
        elif segunda_figura == 'T' or segunda_figura == 't':
            figura = "Triângulo"
            return figura
        elif segunda_figura == 'P' or segunda_figura == 'p':
            figura = "Trapézio"
            return figura
        elif segunda_figura == 'C' or segunda_figura == 'c':
            figura = "Círculo"
            return figura

        

def calcular_area_circulo(pi,raio):
    area = pi*raio**2
    return area


def calcular_area_quadrado(lado_quadrado):
    area = lado_quadrado**2
    return area


def calcular_area_retangulo(comprimento_retangulo,largura_retangulo):
    area = comprimento_retangulo*largura_retangulo
    return area


def calcular_area_triangulo(base_triangulo,altura):
    area = (base_triangulo*altura)/2
    return area


def calcular_area_trapezio(base_maior_trapézio,base_menor_trapézio,altura_trapezio):
    area = ((base_maior_trapézio+base_menor_trapézio)*altura_trapezio)/2


def obter_numero_positivo (label = "Digite um número positivo: "):
    numero = input(label)
    while not numero.isnumeric() or numero == '' or numero<=0:
        print ("Valor inválido!")
        numero = input(label)
        return int(numero)


def obter_numero (label = "Digite um número: "):
    numero = input(label)
    while not numero.isnumeric() or numero == '':
        print ("Valor inválido!")
        numero = input(label)
        return int(numero)
    

def obter_texto(label = "Digite uma string: "):
    texto = input(label)
    return texto


def mostrar_texto(texto):
    print(texto)

main()

File: API-MS_28jun2023/GABRIEL DA SILVA LEAL/prova/areas.js
import prompt from 'prompt-sync'
let input = prompt()

function main(){
    
    mostrar_opcoes()
    let figura_1 = input('Digite a letra correspondente a  primeira figura: ').toUpperCase()
    let figura_2 = input('Letra correspondente a segunda figura: ').toUpperCase()

    if(figura_1 !== figura_2){

        calculando_area_das_figuras(figura_1)
        calculando_area_das_figuras(figura_2)

        

    
    }

    

}
main()



function mostrar_opcoes(){
    let opcoes = 'Q- Quadrado' + '\n' + 'R- Retângulo' + '\n' + 'T- Triângulo' + '\n' + 'C- Círculo' + '\n' + 'Z- Trapézio'

    console.log(opcoes)
}



function calculando_area_das_figuras(figura1){
    let area = 0
    const PI = 3.14159

    if(figura1 == 'Q'){

        let lado = Number(input('Digite o lado do quadrado: '))

        if(lado < 0){

            lado =  Number(input('Digite o valor novamente: '))
        }

        area = lado ** 2

        console.log(`A área do Quadrado é de ${area}`)
    }

    else if(figura1 == 'R'){

        let largura = Number(input('Digite a largura do retângulo: '))
        let comprimento = Number(input('Digite o comprimento do retângulo: '))

        if(largura < 0){

        let largura = Number(input('Digite a largura novamente: '))
    
        }

        if(comprimento < 0){

            let comprimento = Number(input('Digite comprimento novamente: '))
        
            }
        
        area = largura * comprimento 

        console.log(`A área do Retângulo é de ${area}`)
    }

    else if(figura1 == 'T'){

        let base = Number(input('Digite a base do triângulo: '))
        let altura = Number(input('Digite a altura do triângulo: '))

        if(base < 0){

            let base = Number(input('Digite a base novamente: '))
        
            }
    
        if(altura < 0){
    
            let comprimento = Number(input('Digite a altura novamente: '))
            
            }


        area = (base * altura) / 2

        console.log(`A área do Triângulo é de ${area}`)
    }

    else if(figura1 == 'Z'){

        let altura = Number(input('Digite a altura do trapézio: '))
        let base_maior = Number(input('Digite o comprimento da base maior: '))
        let base_menor = Number(input('Digite o comprimento da base menor: '))

        if(altura < 0){

            altura = Number(input('Digite a altura novamente: '))
        }
        if(base_maior < 0){

            let base = Number(input('Digite a base maior novamente: '))
        
            }
    
        if(base_menor < 0){
    
            let comprimento = Number(input('Digite a base menor novamente: '))
            
            }


        
        area = ((base_maior + base_menor) * altura) / 2
        
        console.log(`A área do Trapézio é de ${area}`)
    }

    else if(figura1 == 'C'){

        let raio = Number(input('Digite o raio do cŕculo: '))

        if(raio < 0 ){

            raio = Number(input('Digite o raio novamente: '))
        }


        area = PI * (raio ** 2)
        console.log(`A área do Círculo é de ${area}`)
    }

    let area2 = 0 
   
}


function descobrindo_maior_area(area_figura1, area_figura2){

    if(area_figura1 > area_figura2){

        console.log('A área da figura 1 é maior que a da figura 2!')

    }else{

        console.log('A área da figura 2 é maior que a da figura 1!')
    }
}

